projects/plugins/jetpack/extensions/blocks/writing-prompts/editor.js

```diff
@@ -0,0 +1,27 @@
+import { createBlock } from '@wordpress/blocks';
+import { dispatch } from '@wordpress/data';
+import { waitForEditor } from '../../shared/wait-for-editor';
+
+async function insertTemplate( prompt, embedPrompt = false ) {
+	await waitForEditor();
+
+	const { insertBlocks } = dispatch( 'core/block-editor' );
+	const writingPromptBlocks = embedPrompt
+		? [ createBlock( 'core/pullquote', { value: prompt.text } ), createBlock( 'core/paragraph' ) ]
+		: createBlock( 'core/paragraph', { placeholder: prompt.text }, [] );
+
+	insertBlocks( writingPromptBlocks, 0, undefined, false );
+}
+
+function initWritingPrompts() {
+	const data = window.Jetpack_WritingPrompts;
+	const urlQuery = new URLSearchParams( document.location.search );
+	const embedPrompt = !! urlQuery.get( 'embed_prompt' );
+	if ( typeof data !== 'object' || ! data.prompts || data.prompts.length < 1 ) {
```

Comment:
nit:

I just checked in my browser, if `data.prompts.length` were undefined, `data.prompts.length < 1` would return `false`, better to check for `data.prompts[ 0 ]` directly
```suggestion
	if ( typeof data !== 'object' || ! data.prompts || ! data.prompts[ 0 ] ) {
```

Reply:
Thanks for catching--updated.